updater/updating.go

```diff
@@ -60,6 +60,12 @@ func (reg *ResourceRegistry) downloadIndex(idx Index) error {
 		log.Warningf("%s: failed to add resources: %s", reg.Name, err)
 	}
 
+	// check if dest dir exists
+	err = reg.storageDir.EnsureRelPath(filepath.Dir(idx.Path))
+	if err != nil {
+		log.Warningf("%s: failed to ensure directory for updated index %s: %s", reg.Name, idx.Path, err)
+	}
+
```

Comment:
In case that fails, isn't `ioutil.WriteFile` below supposed to fail too? Shouldn't we do an early exit here?

Reply:
`EnsureRelPath` does more than just creating the needed directly. In order to retain resilience, I'd rather have WriteFile fail, it is also might succeed, even if something was not perfectly right before.